eks_infra/create_cluster.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+#/ Usage: 
+#/ export AWS_REGION=<AWS-Region>
+#/ export EC2_KEY_PAIR_NAME=<EC2-Key-Pair-Name>
+#/ ./create_cluster.sh eks_cluster_name eks_version
+
+set -e
+
+# Function to create EC2 key pair
+function create_ec2_key_pair() {
+    aws ec2 create-key-pair \
+    --key-name ${1} \
+    --query 'KeyMaterial' \
+    --output text > ./${1}-KeyPair.pem
+}
+
+# Function to create EKS cluster using eksctl. 
+# The cluster name follows the {framework}-{build_context} convention
+function create_eks_cluster() {
+
+    if [ "${3}" = "us-east-1" ]; then
+      ZONE_LIST=(a b d)
+    else
+      ZONE_LIST=(a b c)
+    fi
+    
+    eksctl create cluster \
+    --name ${1} \
+    --version ${2} \
+    --zones=${3}${ZONE_LIST[0]},${3}${ZONE_LIST[1]},${3}${ZONE_LIST[2]} \
+    --without-nodegroup
+}
+
+# Function to create static and dynamic nodegroups in EKS cluster
+function create_node_group(){
+    # static nodegroup
+    eksctl create nodegroup \
+    --name static-nodegroup-${2/./-} \
+    --cluster ${1} \
+    --node-type m5.large \
+    --nodes 1 \
+    --node-labels "static=true" \
+    --tags "k8s.io/cluster-autoscaler/node-template/label/static=true" \
+    --asg-access \
+    --ssh-access \
+    --ssh-public-key "${3}"
+
+    # dynamic gpu nodegroup
+    eksctl create nodegroup \
+    --name gpu-nodegroup-${2/./-} \
+    --cluster ${1} \
+    --node-type p3.16xlarge \
+    --nodes-min 0 \
+    --nodes-max 100 \
+    --node-volume-size 80 \
+    --node-labels "test_type=gpu" \
+    --tags "k8s.io/cluster-autoscaler/node-template/label/test_type=gpu" \
+    --asg-access \
+    --ssh-access \
+    --ssh-public-key "${3}"
+
+    # dynamic inf nodegroup
+
+    eksctl create nodegroup \
+    --name inf-nodegroup-${2/./-} \
+    --cluster ${1} \
+    --node-type inf1.xlarge \
+    --nodes-min 0 \
+    --nodes-max 100 \
+    --node-volume-size 500 \
+    --node-ami "ami-092059396c7e51f52" \
+    --node-labels "test_type=inf" \
+    --tags "k8s.io/cluster-autoscaler/node-template/label/test_type=inf" \
+    --asg-access \
+    --ssh-access \
+    --ssh-public-key "${3}"
+  
+  
+}
+
+# Function to create namespaces in EKS cluster
+function create_namespaces(){
+  kubectl create -f namespace.yaml
+}
+
+# Check for input arguments
+if [ $# -ne 2 ]; then
+    echo "${0}: usage: ./create_cluster.sh eks_cluster_name eks_version"
+    exit 1
+fi
+
+# Check for IAM role environment variables
+if [ -z "${AWS_REGION}" ]; then
+  echo "AWS region not configured"
+  exit 1
+fi
+
+CLUSTER=${1}
+EKS_VERSION=${2}
+REGION=${AWS_REGION}
```

Comment:
nit: Unnecessary to copy this into another variable. The script can just use `AWS_REGION`, because it is a required parameter without which the script will exit with error.

Reply:
done